src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -2330,64 +2331,36 @@ public async Task<JArray> GetObjectValues(int objectId, GetObjectCommandOptions
                             },
                             name = "Target"
                         });
-                ret.Add(obj);
-                return ret;
+                objectValues.Add(obj);
+                return objectValues;
             }
-            for (int i = 0; i < typeId.Count; i++)
+            for (int i = 0; i < typeIdsIncludingParent.Count; i++)
```

Comment:
This whole thing can be simplified, I think, in a future PR. Something like:

```csharp

            for (int i = 0; i < typeIdsIncludingParent.Count; i++)
            {
                int typeId = typeIdsIncludingParent[i];
                // 0th id is for the object itself, and then its parents
                bool isOwn = i == 0;

                if (!getCommandType.HasFlag(GetObjectCommandOptions.AccessorPropertiesOnly) || isOwn)
                {
                    // !AccessorPropertiesOnly: add *all* fields -
                    //  AccessorPropertiesOnly: add isOwn fields
                    await AddDebuggerBrowsableFieldValuesFor(typeId, isOwn, objectValues);
                }

                if (getCommandType.HasFlag(GetObjectCommandOptions.WithProperties))
                    await AddDebuggerBrowsablePropertyValuesFor(typeId, isOwn, objectValues);

                // ownProperties
                // Note: ownProperties should mean that we return members of the klass itself,
                // but we are going to ignore that here, because otherwise vscode/chrome don't
                // seem to ask for inherited fields at all.
                //if (ownProperties)
                    //break;
                /*if (accessorPropertiesOnly)
                    break;*/
            }
```

Reply:
You are right, but I would prefer to do it in this PR. I will keep it open till I finish with properties.